plugins/xsettings/msd-xsettings-manager.c

```diff
@@ -76,12 +76,14 @@
 #define DPI_HIGH_REASONABLE_VALUE 500
 
 /* The minimum resolution at which we turn on a window-scale of 2 */
-#define HIDPI_LIMIT (DPI_FALLBACK * 2)
+/* Set this to 90% of 2x DPI_FALLBACK, to catch QHD laptop screens */
+/* that are just below the 2x DPI_FALLBACK mark */
+#define HIDPI_LIMIT (DPI_FALLBACK * 2 * 90 / 100)
```

Comment:
I did not test this PR, but if it will be merged just a comment:
90/100 means integer division, so this results in 0. Maybe just change it to 0.9 and 2 to 2.0.

Reply:
@mbkma I don't think that's how the compiler works: it process `*` and `/` operators in the order they appear, so the first value being `int`, it ends up with an `int`:

```
$ cat test.c 
#include <stdio.h>

#define DPI_FALLBACK 96
#define HIDPI_LIMIT (DPI_FALLBACK * 2 * 90 / 100)

int main() {
  printf("%d\n", HIDPI_LIMIT);
}
$ gcc test.c && ./a.out
172
```

So the final value is 172.

Reply:
ok, but I still consider that bad style. First you rely on the value before the `/` being bigger than `100` for this to work. Second the `HIDPI_LIMIT` constant will is compared to a `double` anyway at line `272` requiring an implicit cast by the compiler of `HIDPI_LIMIT` to double. Also you have one (unnecessary) additional division.

Reply:
> First you rely on the value before the / being bigger than 100

Not sure I understand. Eg: if `DPI_FALLBACK` is 25, then it'll be `25 * 2 * 90` = `4500` which then `4500 / 100` is `45`, so... it works.

> the HIDPI_LIMIT constant is compared to a double anyway at line 272 requiring an implicit cast by the compiler of HIDPI_LIMIT to double

Yes, thats how the code was before this PR: `DPI_FALLBACK`, `DPI_LOW_REASONABLE_VALUE`, `DPI_HIGH_REASONABLE_VALUE`, `HIDPI_LIMIT` are all ints, and they're used in maths with double. TBH, I'd not be surprised if the compiler actually generates doubles from these as an optimization.

> Also you have one (unnecessary) additional division.

I think it'll not be productitive if I try to guess exctly how you think the code should be. I can change this to:

```c
#define HIDPI_LIMIT ((double)(DPI_FALLBACK) * 2.0 * 0.9)
```

but then an argument could be made about having 2 multiplications, a cast to `DPI_FALLBACK` which is an `int` or... something I can't foresee.

May I suggest @mbkma that you try this:
![image](https://user-images.githubusercontent.com/1386232/189741613-a8d35fe7-3fa5-4eda-9987-720c2d1f6dbd.png)
and share **exactly** how you prefer it to be, and I'll make it so? I just want to move this 2 line change forward.

Reply:
I don't want to block this. The only thing I am suggesting is replacing `90/100` by `0.9`. But either way is fine.